.github/workflows/gradle.publish.yaml

```diff
@@ -146,6 +113,20 @@ jobs:
           onlyLastTag: true
           maxIssues: ${{ inputs.changelog_max_issues }}
 
+      - id: generate-release-information
+        name: "Generate release information"
+        run: |
+          release_version_suffix = "alpha"
+          if [[ ${{ inputs.curse_release_type }} == release ]]; then
+            release_version_suffix = ""
+          elif [[ ${{ inputs.curse_release_type }} == beta ]]; then
+            release_version_suffix = "snapshot"
+          fi
+          echo "Release version suffix: ${release_version_suffix}"
+          echo "release_version_suffix=${release_version_suffix}" >> "$GITHUB_ENV"
+          ./gradlew # Prevent download message from appearing inside the getProjectName output
+          echo "mod_name=$(./gradlew --build-cache getProjectName)" >> "$GITHUB_ENV"
```

Comment:
I understand why you would want to do this, but instead of doing this here, I would rather you do this logic of the curse file name entirely in gradle and not in the action.

Reply:
Specifically because things like the project name are known, and you can trivially compute the primary version section, by joining MOD_VERSION and the minecraftVersion property in OPC, as well as the CURSE_RELEASE_TYPE, without altering any versioning logic, or requiring this extra jump through hoops.